### ISSUE TITLE
Catch-22 with Fedora and BTRFS

### ISSUE THREAD

**Issue by ric-m** [linuxmint/timeshift#368]
**Describe the bug**
Cannot restore because filesystem doesn't conform to Ubuntu layout standard where prefixes use @

**To Reproduce**
1. Install Fedora + Timeshift
2. Volume is BTRFS by default
3. Snapshots created perfectly
4. Error message: cannot restore: Unsupported subvolume layout

**Expected behavior**
Warnings about BTRFS have come true: behaviour is exactly as predicted

**Wanted behavior**
1. Want to create snapshots on separate disk formatted as EXT4 as advised in the readme.
2. therefore have to use rsync (btrfs mode does not permit different device)
3. different device important in case source drive damaged/corrupted/etc
4. snapshots created exactly as expected
5. cannot restore because filesystem subvolumes are not prefixed with @
6. the catch: cannot use btrfs mode because i am using separate internal disk; cannot restore using rsync method because filesystem does not conform to ubuntu standard.
7. between a rock and a hard place and no idea how to resolve (_really_ don't want to mess with fedora filesystem)
8. Is there a way forward or is Timeshift just not for me?

**Screenshots**
n/a

**System:**
 - OS: Fedora 41
 - Desktop: KDE Plasma 6
 - Application Version: Timeshift 21.11.2

**Comment by 10leej** [comment]
@clefebvre this has been a known issue with timeshift basically ever since it added btrfs support.
Will this ever be fixed?

**Comment by ErenYanic** [comment]
I don't think it's a very difficult problem to solve, but it seems that timeshift developers don't care

**Comment by ric-m** [comment]
Agreed. I appreciate the OS is immutable but being able to roll-back apps installed on top of the OS to a previous stable state is surely a requirement of every modern OS.

I abandoned timeshift after it completely bricked a perfectly working system during tests.

**Comment by ErenYanic** [comment]
I need a timeshift-like application for my fedora system, I think I will search the fedora forums for solutions developed/used by other fedora users.

**Comment by 10leej** [comment]
> I need a timeshift-like application for my fedora system, I think I will search the fedora forums for solutions developed/used by other fedora users.

Snapper

**Comment by ErenYanic** [comment]
> > I need a timeshift-like application for my fedora system, I think I will search the fedora forums for solutions developed/used by other fedora users.
> 
> Snapper

Thanks, this is another alternative, I guess we should try them. https://gitlab.com/btrfs-assistant/btrfs-assistant

**Comment by svenjacobs** [comment]
I cannot even setup Timeshift on Fedora 42 with a default BTRFS file system layout because _afaik_ Timeshift expects the root subvolume to be named `@` but it's `root` on Fedora. Please make Timeshift compatible with non-Ubuntu-like distributions 😃

**Comment by ErenYanic** [comment]
> I cannot even setup Timeshift on Fedora 42 with a default BTRFS file system layout because _afaik_ Timeshift expects the root subvolume to be named `@` but it's `root` on Fedora. Please make Timeshift compatible with non-Ubuntu-like distributions 😃

In Linux Mint, timeshift developers don't care about other linux distributions, this problem has existed from the beginning and they haven't solved it. This is not the first issue thread about this problem, but it seems that they do not care about us linux users from other distributions. That's why I don't recommend using timeshift, there are very good alternatives and we don't have to use timeshift.

I have been using the btrfs assistant application developed by Garuda Linux developers for 2-3 months and it works smoothly and there is no problem or unsupported problem caused by distribution differences. In addition, the Déjà Dup Backups application developed by fedora can be another alternative.

https://gitlab.com/btrfs-assistant/btrfs-assistant

**Comment by svenjacobs** [comment]
> In Linux Mint, timeshift developers don't care about other linux distributions, this problem has existed from the beginning and they haven't solved it. This is not the first issue thread about this problem, but it seems that they do not care about us linux users from other distributions. That's why I don't recommend using timeshift, there are very good alternatives and we don't have to use timeshift.

That's very sad.

> I have been using the btrfs assistant application developed by Garuda Linux developers for 2-3 months and it works smoothly and there is no problem or unsupported problem caused by distribution differences.

I tried BTRFS Assistant but it wants me to create a `snapper` configuration first. It was not very intuitive and I currently have no experience with `snapper`. I guess I need to read about `snapper` first. I tried Timeshift in a Linux Mint virtual machine and the setup was pretty easy and straightforward.

> In addition, the Déjà Dup Backups application developed by fedora can be another alternative.

I'm explicitly looking for a backup solution that utilizes BTRFS snapshots to create regular snapshots of the root subvolume in case some package updates break system configuration or so. For backing up my home folder on a NAS I already use [Kopia](https://kopia.io).

**Comment by ErenYanic** [comment]
> > In Linux Mint, timeshift developers don't care about other linux distributions, this problem has existed from the beginning and they haven't solved it. This is not the first issue thread about this problem, but it seems that they do not care about us linux users from other distributions. That's why I don't recommend using timeshift, there are very good alternatives and we don't have to use timeshift.
> 
> That's very sad.
> 
> > I have been using the btrfs assistant application developed by Garuda Linux developers for 2-3 months and it works smoothly and there is no problem or unsupported problem caused by distribution differences.
> 
> I tried BTRFS Assistant but it wants me to create a `snapper` configuration first. It was not very intuitive and I currently have no experience with `snapper`. I guess I need to read about `snapper` first. I tried Timeshift in a Linux Mint virtual machine and the setup was pretty easy and straightforward.
> 
> > In addition, the Déjà Dup Backups application developed by fedora can be another alternative.
> 
> I'm explicitly looking for a backup solution that utilizes BTRFS snapshots to create regular snapshots of the root subvolume in case some package updates break system configuration or so. For backing up my home folder on a NAS I already use [Kopia](https://kopia.io).

Btrfs assistant seems quite complicated, yes because it has a few more features other than just taking snapshots, but I don't use those features. At first, I figured it out by reading, tinkering and asking large language models a bit with the help thread someone opened for btrfs assistant in a fedora forum and other users guiding him. 

I think if you give it a little more chance and tinker with it, it will run smoothly on fedora system and you can get rid of this problem. I can also help you if you want, because a few months ago I was looking for a solution to this problem. I solved it with the Btrfs Assistant application with the advice of someone I know (he does not use fedora but he is experienced in linux) and a little research as I mentioned in the first paragraph.

As a normal user, I take a snapshot of my own system into the same ssd.

**Comment by ric-m** [comment]
> I'm explicitly looking for a backup solution that utilizes BTRFS snapshots to create regular snapshots of the root subvolume in case some package updates break system configuration or so. For backing up my home folder on a NAS I already use [Kopia](https://kopia.io).

Hello,
I started this thread and I too had high hopes of finding a solution.

I tried to get Timeshift to work with Kinoite (immutable OS) but that failed. I tried it with Fedora 40 ( by manually creating partitions to comply with the @name requirement) and Timeshift mangled the boot files on a test restore. I figured that Timeshift on fedora was just too risky. I abandoned Timeshift at that point.

I then moved over to Flatpacks but since I first posted I realise that not all flatpacks are maintained. I used a utility - Warehouse - that I hoped would help me manage applications including roll-backs and upgrades. Unless it's mainstream, flatpacks are either just not available or not maintained. I haven't used Warehouse in months.... In fact, it's about time i uninstalled it.

I tried SNAP for one particular package not available by any other means. Hated it, uninstalled the package and uninstalled SNAP. the residues of that particular adventure are probably still littering my hard drive.

The ONLY reliable installer I've found is dnf. I have a few residual flatpacks that I shall migrate to dnf and then I shall have _one unified approach_. I cannot stress the benefits of a single approach enough.

I feel your pain. And disappointment. I've been there too. For me the bottom line is there just ain't anything that manages applications and upgrades, reverts via a reliable GUI.

At some point I think immutable OS installs will become the standard. But on top of that there will be need for a reliable installer/uninstaller/updgrader/downgrader that can manage applications. I think we're some way from that.

Until then I have reverted to command line instructions. The very last thing I wanted, but it works.

Ric
